fl_training.py

```diff
@@ -38,88 +36,110 @@ def preprocess_node_list(node_list):
         node.y_train = y_node_train
         node.y_val = y_node_val
         
-        print('Node #' + str(node_index) + ': done.')
-        
+        print('   Node #' + str(node_index) + ': done.')
+    
+    print('   Done.')    
     return node_list
 
 
+#%% Pre-process test data for model evaluation
+    
+def preprocess_test_data(x_test, y_test):
+    """ Return x_test and y_test preprocessed for keras CNN"""
+    
+    print('\n## Pre-processing test data for keras CNN:')
+    x_test = utils.preprocess_input(x_test)
+    y_test = keras.utils.to_categorical(y_test, constants.NUM_CLASSES)
+    
+    print('   Done.')
+    return x_test, y_test
+
+
 #%% Single partner training
     
 def compute_test_score_for_single_node(node, epoch_count):
     """Return the score on test data of a model trained on a single node"""
     
+    print('\n## Training and evaluating model on one single node.')
     # Initialize model
     model = utils.generate_new_cnn_model()
     # print(model.summary())
 
     # Train model
-    print('\n### Training model on one single node: node ' + node.node_id)
+    print('\n### Training model on one single node: ' + str(node))
```

Comment:
This is the output on my machine:
> ### Training model on one single node: <node.Node object at 0x000001F23B351F60>
Because Node class does not have` __str__` or `__repr__` methods implemented --> https://stackoverflow.com/questions/4932438/how-to-create-a-custom-string-representation-for-a-class-object